lib/measured/unit.rb

```diff
@@ -56,8 +56,8 @@ def inverse_conversion_amount
 
   private
 
-  def case_insensitive(comparison)
-    [comparison].flatten.map(&:downcase)
+  def downcased_names
+    @downcased_names ||= @names.map(&:downcase)
```

Comment:
For a lot of items, definitely a `Set` is a good option, but for a small number of items an `Array` is fine:

```ruby
Benchmark.ips do |x|
  ARRAY = %w(lb lbs pound pounds)
  SET = Set.new(ARRAY)
  x.report("array") { ARRAY.include?('pounds') }
  x.report("set") { SET.include?('pounds') }
  x.compare!
end
```
```
Warming up --------------------------------------
               array   221.743k i/100ms
                 set   222.429k i/100ms
Calculating -------------------------------------
               array      5.455M (± 6.5%) i/s -     27.274M in   5.023466s
                 set      5.347M (± 5.1%) i/s -     26.691M in   5.004617s

Comparison:
               array:  5455255.6 i/s
                 set:  5347423.9 i/s - same-ish: difference falls within error
```

```ruby
Benchmark.ips do |x|
  ARRAY = %w(g gram grams)
  SET = Set.new(ARRAY)
  x.report("array") { ARRAY.include?('grams') }
  x.report("set") { SET.include?('grams') }
  x.compare!
end
```
```
Warming up --------------------------------------
               array   232.450k i/100ms
                 set   220.728k i/100ms
Calculating -------------------------------------
               array      5.750M (± 6.4%) i/s -     28.824M in   5.033535s
                 set      5.202M (± 6.1%) i/s -     26.046M in   5.027608s

Comparison:
               array:  5750476.0 i/s
                 set:  5201677.4 i/s - same-ish: difference falls within error
```

There's generally one of two reasons this happens:

1. They are implemented with hashing, so the item being checked for inclusion has to go through a hashing function, and then a (more) complicated lookup, or
2. They are implemented as a tree structure, in which case they often lose memory characteristics that have terrible cache characteristics.

I think I'd rather stick to an array for now.